source/common/http/utility.cc

```diff
@@ -136,17 +136,27 @@ bool Utility::isInternalRequest(const HeaderMap& headers) {
   return Network::Utility::isInternalAddress(forwarded_for->value().c_str());
 }
 
-uint64_t Utility::parseCodecOptions(const Json::Object& config) {
-  uint64_t ret = 0;
+Http2Settings Utility::parseHttp2Settings(const Json::Object& config) {
+  Http2Settings ret;
+
   std::string options = config.getString("http_codec_options", "");
   for (const std::string& option : StringUtil::split(options, ',')) {
     if (option == "no_compression") {
-      ret |= CodecOptions::NoCompression;
+      ret.hpack_table_size_ = 0;
     } else {
       throw EnvoyException(fmt::format("unknown http codec option '{}'", option));
     }
   }
 
+  Json::ObjectSharedPtr http2_settings = config.getObject("http2_settings", true);
+  ret.hpack_table_size_ = http2_settings->getInteger(
+      "hpack_table_size",
```

Comment:
i made 'http2_settings.hpack_table_size' silently overrides 'http_codec_options.no_compression'
here to save me some checking logic, and thought that overrides may also be ok.

i changed it to error out if the two configs conflicts. 

But i can't log warning DEPRECATED yet, since i have somehow get a logger. 
I can pass in a logger from calling site, but may lose the CONFIG logger enum info.